process_report/tests/unit/invoices/test_invoice_fetch.py

```diff
@@ -0,0 +1,35 @@
+import pandas as pd
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from pytest_mock.plugin import MockerFixture
+
+from process_report.tests import util as test_utils
+
+
+def test_fetch_with_mock_s3_bucket(mocker: "MockerFixture") -> None:
```

Comment:
Regardless if the test is compatible for `unittest` or `pytest`, I would suggest you use `unittest.mock` to mock the S3 bucket and `read_csv`, since that is what we use in the rest of this repo. This will also avoid the need to add another dependancy